### DISCUSSION TITLE
How to specify custom elixir language server path

### DISCUSSION THREAD

**Discussion by Pilgrim1379** [LazyVim/LazyVim#1014] [Q&A] [answered]
Thanks for the newly added elixir config. The one issue for me is I prefer to install my language server independenly so I have `mason = false` for al the servers. 

For elixirls, I compile the server myself and its located at `~/lang_servers/elixir-ls/release` as `language_server.sh`.
How do I make the elixir config aware and to use that instead of the mason version?

**Comment by folke** [comment]
The same, just override mason = false for elixir in your config

**Reply by Pilgrim1379** [reply]
I already have but now when I open an elixir file I get a message like `cmd not defined for elixirls, manually call setup cmd in the setup {} call ...` Sorry the rest of the message disappers before I can read all of it. Slow reader :)

I seems like its looking for the elixir language server by the name `elixirls`. However on my system it's `language_server.sh`

**Reply by dpetka2001** [reply, marked as answer]
I believe that you should set the `cmd` by yourself as also stated in the [lspconfig docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#elixirls) to the path that your `language_server.sh` exists.

**Reply by Pilgrim1379** [reply]
I have the following in `plugins\lsp.lua`
```lua
...
servers = {
...
   elixirls = {mason = false},
...
}
```

and in `plugins\elixir.lua`, I have 
```lua
return {
  {
      config = function()
	  require('lspconfig').elixirls.setup {
	      cmd = { "/Users/user/github/language-servers/elixir-ls/release/language_server.sh" },
	  }
	end
   }
}
```
It's however not doing the trick and I'm wondering what I'm doing wrong.

**Reply by dpetka2001** [reply]
I believe your `elixir.lua` is an empty spec without the url of the plugin you want to configure it with. I don't think it's correct (although I might be wrong). Why don't you put the `cmd` in the file `lsp.lua` under `servers.elixirls`?
